build.sbt

```diff
@@ -74,3 +78,19 @@ lazy val relate = Project(
   )
 ).configs(Benchmark).settings(inConfig(Benchmark)(Defaults.testSettings): _*).
   configs(Regression).settings(inConfig(Regression)(Defaults.testSettings): _*)
+
+lazy val macroSettings = buildSettings ++ Seq(
+  moduleName := "relate-macros",
+  scalacOptions += "-language:experimental.macros",
+  addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
+  libraryDependencies <+= (scalaVersion)("org.scala-lang" % "scala-reflect" % _),
+  libraryDependencies ++= Seq(
+    "com.chuusai" %% "shapeless" % "2.3.1" % "test",
```

Comment:
just in tests. `illTyped()` is a great function for testing for compilation failure :)